lib/ui/bottom_bar/pages/bottom_bar.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:news_pulse/ui/home/pages/home_page.dart';
+import 'package:persistent_bottom_nav_bar/persistent_tab_view.dart';
+
+import '../../../core/di/di_manager.dart';
+
+class BottomBar extends StatefulWidget {
+  const BottomBar({super.key});
+  static const String routeName = "/bottom-bar";
+
+  @override
+  State<BottomBar> createState() => _BotttomBarState();
+}
+
+class _BotttomBarState extends State<BottomBar> {
+  final PersistentTabController _controller =
+      PersistentTabController(initialIndex: 0);
+  @override
+  Widget build(BuildContext context) {
+    return PersistentTabView(
+      context,
+      controller: _controller,
+      screens: _buildScreens(),
+      items: _navBarsItems(),
+      confineInSafeArea: true,
+      backgroundColor: DIManager.findCC().primaryColor,
+      handleAndroidBackButtonPress: true, // Default is true.
+      resizeToAvoidBottomInset:
+          true, // This needs to be true if you want to move up the screen when keyboard appears. Default is true.
+      stateManagement: true, // Default is true.
+      hideNavigationBarWhenKeyboardShows:
+          true, // Recommended to set 'resizeToAvoidBottomInset' as true while using this argument. Default is true.
+      popAllScreensOnTapOfSelectedTab: true,
+      popActionScreens: PopActionScreensType.all,
+      itemAnimationProperties: const ItemAnimationProperties(
+        // Navigation Bar's items animation properties.
+        duration: Duration(milliseconds: 200),
+        curve: Curves.ease,
+      ),
+      screenTransitionAnimation: const ScreenTransitionAnimation(
+        // Screen transition animation on change of selected tab.
+        animateTabTransition: true,
+        curve: Curves.ease,
+        duration: Duration(milliseconds: 200),
+      ),
+      navBarStyle:
+          NavBarStyle.style6, // Choose the nav bar style with this property.
+    );
+  }
+}
+
+List<Widget> _buildScreens() {
+  return [const HomePage(), Container()];
+}
+
+List<PersistentBottomNavBarItem> _navBarsItems() {
+  return [
+    PersistentBottomNavBarItem(
+      icon: const Icon(CupertinoIcons.home),
+      title: ("Home"),
+      activeColorPrimary: CupertinoColors.activeBlue,
```

Comment:
use DIManager.findCC not CupertinoColors 
btw Cupertino is a lib for ios not android